assets/js/googlesitekit/datastore/ui/ui.test.js

```diff
@@ -0,0 +1,154 @@
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { createTestRegistry } from 'tests/js/utils';
+import { STORE_NAME } from './constants';
+
+describe( 'core/ui store', () => {
+	let registry;
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	describe( 'actions', () => {
+		describe( 'setValues', () => {
+			it( 'requires the values param', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues();
+				} ).toThrow( 'values must be an object.' );
+			} );
+
+			it( 'requires the values param to be an object not an array', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues( [] );
+				} ).toThrow( 'values must be an object.' );
+			} );
+
+			it( 'does not throw if values is an object', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues( {} );
+				} ).not.toThrow();
+			} );
+
+			it( 'requires the values param to be an object not a string', () => {
+				expect( () => {
+					registry.dispatch( STORE_NAME ).setValues( 'values' );
+				} ).toThrow( 'values must be an object.' );
+			} );
+
+			it( 'does not overwrite unrelated keys', () => {
+				registry.dispatch( STORE_NAME ).setValues( { key1: 'value1', key2: 'value2' } );
+
+				registry.dispatch( STORE_NAME ).setValues( { key1: 'value3' } );
+
+				const formValue = registry.select( STORE_NAME ).getValue( 'key2' );
+				expect( formValue ).toEqual( 'value2' );
+			} );
+
+			it( 'returns a newly-set value if a new value for an existing key is set', () => {
+				registry.dispatch( STORE_NAME ).setValues( { key1: 'value1', key2: 'value2' } );
+
+				registry.dispatch( STORE_NAME ).setValues( { key1: 'value3' } );
+
+				const formValue = registry.select( STORE_NAME ).getValue( 'key1' );
+				expect( formValue ).toEqual( 'value3' );
+			} );
+
+			it( 'does not overwrite unrelated keys when an empty object is supplied to values', () => {
+				registry.dispatch( STORE_NAME ).setValues( { key1: 'value1', key2: 'value2' } );
+
+				registry.dispatch( STORE_NAME ).setValues( {} );
+
+				const formValue = registry.select( STORE_NAME ).getValue( 'key2' );
+				expect( formValue ).toEqual( 'value2' );
+			} );
+
+			it( 'works with empty values where the key value is updated', () => {
+				registry.dispatch( STORE_NAME ).setValues( {} );
+
+				registry.dispatch( STORE_NAME ).setValues( { key1: 'value1', key2: 'value2' } );
+
+				const formValue = registry.select( STORE_NAME ).getValue( 'key2' );
+				expect( formValue ).toEqual( 'value2' );
+			} );
+
+			it( 'preserves data from state when new data is assigned', () => {
+				registry.dispatch( STORE_NAME ).setValues( { key1: 'value1', key2: 'value2' } );
+
+				registry.dispatch( STORE_NAME ).setValues( { key3: 'value3', key4: 'value4' } );
+
+				const formValue = registry.select( STORE_NAME ).getValue( 'key2' );
+				expect( formValue ).toEqual( 'value2' );
+			} );
```

Comment:
I'd add some tests in here with non-string _values_ as I think we want to allow them and it'd be good if the tests explicitly supported that 🙂

Reply:
👍